nautobot/extras/datasources/registry.py

```diff
@@ -1,12 +1,14 @@
 """Registry-related APIs for datasources."""
+from collections import OrderedDict
+
 from nautobot.extras.choices import JobResultStatusChoices, LogLevelChoices
 from nautobot.extras.context_managers import change_logging
 from nautobot.extras.registry import registry
 
 
 def get_datasource_contents(model_name):
     """Get the list of DatasourceContent entries registered for a given model name."""
-    return sorted(registry["datasource_contents"].get(model_name, []))
+    return sorted(registry["datasource_contents"].get(model_name, []), key=lambda datasource: datasource.weight)
```

Comment:
Do we need to use `datasource.name` as a secondary key for the case where multiple datasources have the same weight?